src/core/homepage_elements/journals_and_html/hooks.py

```diff
@@ -0,0 +1,55 @@
+__copyright__ = "Copyright 2017 Birkbeck, University of London"
+__author__ = "Martin Paul Eve & Andy Byers"
+__license__ = "AGPL v3"
+__maintainer__ = "Birkbeck Centre for Technology and Publishing"
+
+import random
+
+from journal import models as journal_models
+from core.homepage_elements.journals_and_html import plugin_settings
+from utils.logger import get_logger
+from press import models as press_models
+
+logger = get_logger(__name__)
+
+
+def get_random_journals():
+    journals = journal_models.Journal.objects.filter(hide_from_press=False)
+    journal_pks = [journal.pk for journal in journals]
+    random_journal_pks = list()
+
+    random.shuffle(journal_pks)
+    for i in range(0, 6):
+
+        if journal_pks:
+            choice = journal_pks.pop()
+            random_journal_pks.append(choice)
+
+    return journals.filter(pk__in=random_journal_pks)
+
+
+def yield_homepage_element_context(request, homepage_elements):
+    if homepage_elements is not None and homepage_elements.filter(
+            name=plugin_settings.PLUGIN_NAME,
+    ).exists():
+
+        if request.press.random_featured_journals:
+            featured_journals = get_random_journals()
+        else:
+            featured_journals = request.press.featured_journals.all()
```

Comment:
I don't understand this bit, isn't this the responsibility of the featured journals homepage element?

Reply:
This HPE also uses featured journals in a side by side with an HTML block. Its an implementation of the hack for Huddersfield's homepage.